dashboard/app/models/user.rb

```diff
@@ -503,23 +503,18 @@ def sanitize_race_data_set_urm
     true
   end
 
-  # Implement validation that refuses to set admin:true attribute unless
-  # there's a Code.org google sso option present.  Unmigrated users are
-  # not allowed to be admins.
+  # Only allow admin permission for studio accounts with Google OAuth authentication.
   validate :enforce_google_sso_for_admin
   def enforce_google_sso_for_admin
     return unless admin
 
     errors.add(:admin, 'must be a migrated user') unless migrated?
 
-    google_oauth = google_oauth_authentications
-    errors.add(:admin, 'must have google_oauth2 as an authentication option') unless google_oauth&.present?
+    errors.add(:admin, 'must have only one authentication option') unless authentication_options.count == 1
 
-    errors.add(:admin, 'email must have code.org domain') unless google_oauth.any?(&:codeorg_email?)
-  end
-
-  def google_oauth_authentications
-    authentication_options&.where(credential_type: AuthenticationOption::GOOGLE)
+    unless authentication_options.all? {|ao| ao.google_oauth2? && ao.codeorg_email?}
```

Comment:
i think we should condense lines 513 & 515 into one validation rather than separate validations, since the two validations currently serve the same purpose